src/chains.rs

```diff
@@ -26,14 +26,21 @@ use crate::device::Device as DeviceAPI;
 use crate::{ContextID, Error, SurfaceAccess, SurfaceInfo, SurfaceType};
 use euclid::default::Size2D;
 use fnv::{FnvHashMap, FnvHashSet};
+use glow as gl;
+use glow::Context as Gl;
+use glow::HasContext;
 use log::debug;
-use sparkle::gl::{self, GLuint, Gl};
 use std::collections::hash_map::Entry;
 use std::fmt::Debug;
 use std::hash::Hash;
 use std::mem;
+use std::num::NonZero;
 use std::sync::{Arc, Mutex, MutexGuard, RwLock, RwLockReadGuard, RwLockWriteGuard};
 
+fn framebuffer(framebuffer: u32) -> Option<gl::NativeFramebuffer> {
+    NonZero::new(framebuffer).map(gl::NativeFramebuffer)
+}
```

Comment:
```suggestion
impl Surface {
    fn framebuffer(&self) -> Option<gl::NativeFramebuffer> {
        NonZero::new(framebuffer).map(gl::NativeFramebuffer)
    }
}
```

Reply:
We cannot remove all usages of framebuffer but https://github.com/servo/surfman/pull/318/commits/5d0d31ca6e20aad15b7ee6a277fe6ccfaddf4931 does most of it.